ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/services/ClaimMigrator.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.cmc.ccd.migration.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.migration.ccd.services.CoreCaseDataService;
+import uk.gov.hmcts.cmc.ccd.migration.idam.models.User;
+import uk.gov.hmcts.cmc.ccd.migration.idam.services.UserService;
+import uk.gov.hmcts.cmc.ccd.migration.repositories.ClaimRepository;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ClaimMigrator {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClaimMigrator.class);
+
+    private ClaimRepository claimRepository;
+    private UserService userService;
+    private CoreCaseDataService coreCaseDataService;
+
+    @Autowired
+    public ClaimMigrator(
+        ClaimRepository claimRepository,
+        UserService userService,
+        CoreCaseDataService coreCaseDataService) {
+        this.claimRepository = claimRepository;
+        this.userService = userService;
+        this.coreCaseDataService = coreCaseDataService;
+    }
+
+    public void migrate() {
```

Comment:
What will happen if in middle of process, one or more create or update in CCD fails for any reason? Will it terminate the process or continue trying for rest of the claims migration? Looks like it will terminate the process. Is that what we want?

Reply:
shouldn't really matter, just keep rerunning till all passed

Reply:
That's fine. just make sure the one it fails on, we don't mark that as migrated.